docs/sphinx/specification/cudaq/types.rst

```diff
@@ -4,30 +4,91 @@ The CUDA Quantum language extension provides certain fundamental types that are
 to quantum-classical computing. These types are provided via defined classical library
 types in C++ in the :code:`cudaq` namespace. 
 
-:code:`cudaq::qudit<Levels>`
-----------------------------
-**[1]** The :code:`cudaq::qudit` models a :math:`D`-level unit of quantum information. The state of
-this system (for :math:`N` qudits) can be described by a :math:`D`\ :sup:`N`\-dimensional vector in
-Hilbert space with the absolute square of all elements summing to 1. 
+Quantum dit (qudit)
+===================
 
-**[2]** The :code:`cudaq::qudit` encapsulates a unique :code:`std::size_t` modeling the index of the
-qudit in the underlying quantum memory space (assuming an infinite register
-of available qudits). 
+The quantum dit, or qudit for short, is a model to represent a :math:`d`-level 
+(or :math:`d`-dimension) quantum system. Mathematically, we describe the state
+of a qudit using a :math:`d`-dimensional vector of unit form, living in the
+vector space :math:`\mathbb{C}^d` (a Hilbert space). Qudits with known values
+for :math:`d` have specific name. A qubit has dimension 2, a qutrit has
+dimension 3, and so on.
 
-**[4]** To adhere to the no-cloning theorem of quantum mechanics,
-the :code:`cudaq::qudit` is non-copyable and non-movable. Therefore, all :code:`cudaq::qudit` 
-instances must be passed by reference, and the no-cloning theorem is satisfied
-at compile-time. 
+In a CUDA Quantum implementation, objects of type qudit must adhere to the
+following constraints:
 
-**[5]** :code:`cudaq::qudit` instances can only be allocated within CUDA Quantum quantum
-kernel code and can never be allocated from classical host code.
+**[1]** Objects of type :code:`qudit` shall be parameterized with :math:`d`,
+        to indicate it's dimension.
 
-**[6]** All instantiated :code:`cudaq::qudit` instances start in the :code:`|0>` computational basis 
-state and serve as the primary input argument for quantum intrinsic operations 
-modeling typical logical quantum gate operations.
+**[2]** CUDA Quantum kernels shall not have objects of type :code:`qudit` with
+        different dimensions.
 
-**[7]** :code:`cudaq::qudit` instances can be negated when leveraged as controls in 
-quantum operation invocations. The mechanism for negation should be via overloading of :code:`qudit<N>::operator!()`.
+**[3]** Objects of type :code:`qudit` shall not carry a quantum state.
+
+**[4]** Objects of type :code:`qudit` shall identify a qudit in the underlying
+        quantum system. The identifier shall be unique and valid during the
+        lifetime of the :code:`qudit` object.
+
+**[5]** Objects of type :code:`qudit` and any directly or indirectly derived
+        type shall only be used within quantum kernel code.
+
+**[6]** The default constructor shall create an object of type code:`qudit` that
+        references a qudit in the :math:`|0\rangle` computational basis state.
+
+**[7]** Objects of type :code:`qudit` may be non-copyable.
```

Comment:
'may' ? or 'are' ? Seems like as-is there is some flexibility - qudits could be copyable if you chose...

Reply:
The definition of the type (and the way we currently use it), says that a `qudit` object is an interface to some portion of the underlying quantum state, and do not carry a quantum state. (A more precise name for this type would be `qudit_ref`.) Threfore, from the perspective of adhering to quantum mechanics quirks, nothing prevents us from allowing something like:
```cpp
cudaq::qudit<2> q0;
cudaq::qubit<2> q1 = q0;
```
The above code just means that both `q0` and `q1` reference the same portion of the underlying quantum state.